tools/federation/src/main/java/org/eclipse/rdf4j/federated/evaluation/concurrent/ParallelTask.java

```diff
@@ -44,4 +44,11 @@ default QueryInfo getQueryInfo() {
 	 * Optional implementation to cancel this task on a best effort basis
 	 */
 	void cancel();
+
+	/**
+	 * Optional implementation to close this task on a best effort basis.
+	 */
+	default void close() {
+
+	}
```

Comment:
I've added a new method to the interface to close a task. Closing a task is similar to cancelling a task except that if a closed task is run then it just becomes a no-op instead of throwing an exception like it does if it is cancelled.